### PR TITLE
Update sdk-feature-store.ipynb to add streaming ingestion

### PR DESCRIPTION
**Pull request by junkourata**
<br>
Add streaming ingestion(write_feature_values) to the feature store notebook.
<br><br><br>

- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

**Comment by ivanmkc** [comment]
@soheilazangeneh PTAL this Feature Store PR

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 1]
```diff
@@ -1281,6 +1281,84 @@
...
     },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Streaming Ingestion\n",
+      "\n",
+      "Streaming ingestion is currently public preview. \n",
+      "\n",
+      "Streaming ingestion lets you make real-time updates to feature values. While batch import is suitable for importing a large volume of data with high latency, streaming ingestion is suitable for ingesting small amount of data with low latency. The written data will be available to read using batch export and online serving. "
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1292]: Rewrite the last sentence as follows (avoid the use of "will"):
"The written data becomes available to read using batch export and online serving."

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 2]
```diff
@@ -1281,6 +1281,84 @@
...
+     ]
+    },
+    {
+     "cell_type": "code",
+     "execution_count": null,
+     "metadata": {},
+     "outputs": [],
+     "source": [
+      "# Since streaming ingestion is public preview, the feature is in aiplatform_v1beta1.\n",
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1301]: "..., the feature is available in..."

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 3]
```diff
@@ -1281,6 +1281,84 @@
...
+      "data_client = FeaturestoreOnlineServingServiceClient(client_options={\"api_endpoint\": API_ENDPOINT})"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "execution_count": null,
+     "metadata": {},
+     "outputs": [],
+     "source": [
+      "# Call write_feature_values to ingest data to users entity_type. \n",
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1323]: Should this be "to the user's entity_type" or "to the user-specified entity_type" or "to the specified entity_type"?

Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1323]: Format write_feature_values as code (`write_feature_values`).

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 4]
```diff
@@ -1281,6 +1281,84 @@
...
+      "    ])"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "execution_count": null,
+     "metadata": {},
+     "outputs": [],
+     "source": [
+      "When `write_feature_values` API succeeded, it returns empty response.\n",
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1358]: Upon successful completion, the `write_feature_values` API returns an empty response.

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 5]
```diff
@@ -1281,6 +1281,84 @@
...
+      "Similary, ingest data to movies entity_type."
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:1359]: Should this be as follows?

"Similarly, ingest data to the entity type `movies_entity_type`."

Note the typo in the spelling of "similarly".

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 1 above)
Comment (sen-sam) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1292]: **Similar changes (suggested rewording by line number):**
261 - Remove "will".
504 - "The movie prediction example lets you create..."
526 - "Running the code cell creates..."
579 - "... A process log is printed."
1028 - "By default, the SDK returns..."
1030 - "By default, all the features of an entity type are selected. The output response displays the selected entity type ID..."
1063 - By default, all the features of an entity type are selected.
1170 - "These are used in the next..."

**Review by sen-sam — CHANGES_REQUESTED** [review]
Additional comments:

- Line 85: Is there any need to surface streaming ingestion on this agenda?
- Line 188: Replace "for executing" with "to execute".

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 1 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1292]: Fixed

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 2 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1301]: Fixed

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 3 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1323]: Fixed

Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1323]: Fixed

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 4 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1358]: We return error code and error message when backend failed to write to storages

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 5 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:1359]: Thanks

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 6]
```diff
@@ -489,19 +490,19 @@
...
         "```\n",
         "* **Featurestore**: the place to store your features\n",
-        "* **Entity type**: under a Featurestore, an Entity type describes an object to be modeled, real one or virtual one.\n",
-        "* **Feature**: under an Entity type, a Feature describes an attribute of the Entity type\n",
+        "* **Entity type**: under a featurestore, an entity type describes an object to be modeled, real one or virtual one.\n",
+        "* **Feature**: under an entity type, a feature describes an attribute of the entity type\n",
         "\n",
-        "In the movie prediction example, you will create a featurestore called `movie_prediction`. This store has 2 entity types: `users` and `movies`. The `users` entity type has the `age`, `gender`, and `liked_genres` features. The `movies` entity type has the `titles`, `genres`, and `average rating` features.\n"
+        "The movie prediction example lets you create a featurestore called `movie_prediction`. This store has 2 entity types: `users` and `movies`. The `users` entity type has the `age`, `gender`, and `liked_genres` features. The `movies` entity type has the `titles`, `genres`, and `average rating` features.\n"
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:505]: "This store has two entity types."

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 7]
```diff
@@ -489,19 +490,19 @@
...
+        "## Terminology and concept\n",
         "\n",
-        "### Featurestore Data model\n",
+        "### Featurestore data model\n",
         "\n",
         "Vertex AI Feature Store organizes data with the following 3 important hierarchical concepts:\n",
         "```\n",
         "Featurestore -> Entity type -> Feature\n",
         "```\n",
         "* **Featurestore**: the place to store your features\n",
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:501]: Lines 501 onwards - capitalize the first word after the colon. E.g. "* **Entity Type**: Under a featurestore..."

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 8]
```diff
@@ -820,9 +823,9 @@
         "id": "K3n5XdK8Xjmw"
       },
       "source": [
-        "## Import Feature Values\n",
+        "## Import feature values\n",
         "\n",
-        "You need to import feature values before you can use them for online/offline serving. In this step, you learn how to import feature values by ingesting the values from Cloud Storage. You can also import feature values from BigQuery or a Pandas dataframe.\n"
+        "You need to import feature values before you can use them for online or offline serving. In this step, you learn how to import feature values by ingesting the values from GCS (Google Cloud Storage). You can also import feature values from BigQuery or a Pandas dataframe.\n"
```
Comment (sen-sam) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:828]: Correct branding: "pandas dataFrame".

**Review by sen-sam — CHANGES_REQUESTED** [review]
Minor changes. Mostly looks great!

**Review by sen-sam — APPROVED** [review]

File: notebooks/official/feature_store/sdk-feature-store.ipynb [hunk 9]
```diff
@@ -84,7 +84,8 @@
         "- Create featurestore, entity type, and feature resources.\n",
```
Comment (soheilazangeneh) [inline notebooks/official/feature_store/sdk-feature-store.ipynb:84]: <p>Should <code>Type</code> be capital in the title? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/954/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

**Review by soheilazangeneh — APPROVED** [review]
LGTM! Left only one comment.

**Comment by andrewferlitsch** [comment]
Notebook needs to be relinted

File: notebooks/official/feature_store/sdk-feature-store.ipynb (hunk 9 above)
Comment (junkourata) [reply notebooks/official/feature_store/sdk-feature-store.ipynb:84]: entity type is actually a correct term. We use it throughout the notebook.

**Pull request merged**